### DISCUSSION TITLE
Allow ui.run outside of the main thread.

### DISCUSSION THREAD

**Discussion by ItsCubeTime** [zauberzeug/nicegui#683] [Ideas / Feature Requests]
Hi 👋 Would be nice if it was possible to run nicegui in a way like:
```py
from nicegui import ui
import asyncio
import time

ui.label("hi")

loop = asyncio.new_event_loop()
loop.run_in_executor(None, lambda: ui.run())
time.sleep(1000)
```
and:
```py
from nicegui import ui
import threading
import time

ui.label("hi")

t = threading.Thread(target=lambda: ui.run(), args = ())
t.daemon = True
t.start()
time.sleep(1000)
```

In my particular case, it would help when wanting to use nicegui together with other UI frameworks that acts as a window host, like Toga.

**Comment by ItsCubeTime** [comment]
It seems that setting `nicegui.ui.run(reload=False)` actually lets the framework run with threading with no modifications as is.

Perhaps, in that case, see if one could offer an alternative to the uvicorn filewatcher that is threading & asyncio compatible?

**Comment by HilkopterBob** [comment]
+1 from me too! would be a great addition without lossing the (much loved) hot reload

**Comment by rodja** [comment]
This would require to do the reloading our self, either with watchfiles (like uvicorn) or hopper. We need to think about that... 
It would be helpful if you could try to restart the thread whenever the filesystem observer detects a change.
